Test Failure: ConcurrencyTestServlet.testWebDDManagedExecutorDefinition LPContextService not available

```
junit.framework.AssertionFailedError: 2022-01-26-04:58:52:216 ERROR: Caught exception attempting to call test method testWebDDManagedExecutorDefinition on servlet test.jakarta.concurrency.web.ConcurrencyTestServlet
javax.naming.NamingException: CWNEN1001E: The object referenced by the java:global/concurrent/dd/web/LPExecutor JNDI name could not be instantiated. If the reference name maps to a JNDI name in the deployment descriptor bindings for the application performing the JNDI lookup, make sure that the JNDI name mapping in the deployment descriptor binding is correct. If the JNDI name mapping is correct, make sure the target resource can be resolved with the specified name relative to the default initial context. [Root exception is com.ibm.wsspi.injectionengine.InjectionException: CWNEN0030E: The server was unable to obtain an object instance for the java:global/concurrent/dd/web/LPExecutor reference.  The exception message was: Unable to find managedExecutorService[java:global/concurrent/dd/web/LPExecutor] for application ConcurrencyTestApp]
	at com.ibm.ws.injectionengine.osgi.internal.naming.InjectionJavaColonHelper.newCannotInstantiateObjectException(InjectionJavaColonHelper.java:183)
	at com.ibm.ws.injectionengine.osgi.internal.naming.InjectionJavaColonHelper.getObjectInstance(InjectionJavaColonHelper.java:118)
	at com.ibm.ws.jndi.url.contexts.javacolon.internal.JavaURLContext.lookup(JavaURLContext.java:332)
	at com.ibm.ws.jndi.url.contexts.javacolon.internal.JavaURLContext.lookup(JavaURLContext.java:370)
	at org.apache.aries.jndi.DelegateContext.lookup(DelegateContext.java:149)
	at java.naming/javax.naming.InitialContext.lookup(InitialContext.java:409)
	at java.naming/javax.naming.InitialContext.doLookup(InitialContext.java:282)
	at test.jakarta.concurrency.web.ConcurrencyTestServlet.testWebDDManagedExecutorDefinition(ConcurrencyTestServlet.java:2228)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at componenttest.app.FATServlet.doGet(FATServlet.java:71)
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:500)
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:587)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1258)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:746)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:443)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1227)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1011)
	at com.ibm.ws.webcontainer.servlet.CacheServletWrapper.handleRequest(CacheServletWrapper.java:75)
	at com.ibm.ws.webcontainer40.servlet.CacheServletWrapper40.handleRequest(CacheServletWrapper40.java:85)
	at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:938)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:281)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1184)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:453)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:412)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:566)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:500)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:360)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready(HttpInboundLink.java:327)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators(NewConnectionInitialReadCallback.java:167)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete(NewConnectionInitialReadCallback.java:75)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:504)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:574)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:958)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1047)
	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:245)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: com.ibm.wsspi.injectionengine.InjectionException: CWNEN0030E: The server was unable to obtain an object instance for the java:global/concurrent/dd/web/LPExecutor reference.  The exception message was: Unable to find managedExecutorService[java:global/concurrent/dd/web/LPExecutor] for application ConcurrencyTestApp
	at com.ibm.wsspi.injectionengine.InjectionBinding.getInjectionObject(InjectionBinding.java:1454)
	at com.ibm.wsspi.injectionengine.InjectionBinding.getInjectionObject(InjectionBinding.java:1384)
	at com.ibm.ws.injectionengine.osgi.internal.naming.InjectionJavaColonHelper.getObjectInstance(InjectionJavaColonHelper.java:116)
Caused by: java.lang.Exception: Unable to find managedExecutorService[java:global/concurrent/dd/web/LPExecutor] for application ConcurrencyTestApp
	at io.openliberty.concurrent.processor.AppDefinedResourceFactory.createResource(AppDefinedResourceFactory.java:94)
	at io.openliberty.concurrent.processor.AppDefinedResourceFactory.createResource(AppDefinedResourceFactory.java:118)
	at com.ibm.ws.injectionengine.osgi.internal.ResourceFactoryObjectFactory.getObjectInstance(ResourceFactoryObjectFactory.java:33)
	at com.ibm.wsspi.injectionengine.InjectionBinding.getInjectionObjectInstance(InjectionBinding.java:1552)
	at com.ibm.wsspi.injectionengine.InjectionBinding.getInjectionObject(InjectionBinding.java:1428)

	at componenttest.topology.utils.FATServletClient.runTest(FATServletClient.java:92)
	at componenttest.custom.junit.runner.SyntheticServletTest.invokeExplosively(SyntheticServletTest.java:42)
	at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:199)
	at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:320)
	at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:173)
```

There is a test case bug, where if tests run in a particular order then a ContextServiceDefinition from the EJB application might not yet be processed, causing managed executors that use it from the web application to fail with the error that is being reported in builds:

```
java.lang.IllegalStateException: CWWKC1201E: MnagedExecutorService java:global/concurrent/dd/web/LPExecutor, which is defined by application ConcurrencyTestApp, requires a ContextService java:global/concurrent/anno/ejb/LPContextService but the ContextService was inaccessible or not available in a timely manner.
```

To avoid this, the test case test case web application can access the EJB upon initialization, making its ContextServiceDefinitions available to all tests in the web app.

--- a/dev/com.ibm.ws.concurrent_fat_jakarta/test-applications/ConcurrencyTestWeb/src/test/jakarta/concurrency/web/ConcurrencyTestServlet.java
+++ b/dev/com.ibm.ws.concurrent_fat_jakarta/test-applications/ConcurrencyTestWeb/src/test/jakarta/concurrency/web/ConcurrencyTestServlet.java
@@ -226,10 +226,11 @@ public class ConcurrencyTestServlet extends FATServlet {
     public void init(ServletConfig config) throws ServletException {
         unmanagedThreads = Executors.newFixedThreadPool(5);
 
-        // This EJB needs to first be used so that its ContextServiceDefinition,
-        // which is relied upon by the other EJB, can be processed
+        // These EJBs need to be used so that their ContextServiceDefinition annotations,
+        // which are relied upon by the web module and the other EJB, can be processed
         try {
             InitialContext.doLookup("java:global/ConcurrencyTestApp/ConcurrencyTestEJB/ContextServiceDefinerBean!test.jakarta.concurrency.ejb.ContextServiceDefinerBean");
+            Executor bean = InitialContext.doLookup("java:global/ConcurrencyTestApp/ConcurrencyTestEJB/ExecutorBean!java.util.concurrent.Executor");
         } catch (NamingException x) {
             throw new ServletException(x);
         }
@@ -3489,10 +3490,6 @@ public class ConcurrencyTestServlet extends FATServlet {
      */
     @Test
     public void testWebMergedManagedScheduledExecutorDefinitionContext() throws Exception {
-        // Ensure annotations from the EJB are processed
-        Executor bean = InitialContext.doLookup("java:global/ConcurrencyTestApp/ConcurrencyTestEJB/ExecutorBean!java.util.concurrent.Executor");
-        assertNotNull(bean);
-
         ManagedScheduledExecutorService executor = InitialContext.doLookup("java:app/concurrent/merged/web/LPScheduledExecutor");
 
         CountDownLatch blocker = new CountDownLatch(1);

